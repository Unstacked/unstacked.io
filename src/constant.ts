export enum Account {
  DEVOPS = '681601794463',
  WORKLOAD = '806124249357',
}
