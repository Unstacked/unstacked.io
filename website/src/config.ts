import type{ NavItems } from './types'

export const NAV_ITEMS: NavItems = {
    home: {
        path: '/',
        title: 'home'
    },
    blog: {
        path: '/blog',
        title: 'blog'
    },
    tags: {
        path: '/tags',
        title: 'tags'
    },
    media: {
        path: '/media',
        title: 'media'
    },
    about: {
        path: '/about',
        title: 'about'
    }
}

export const SITE = {
    // Your site's detail?
    name: 'Unstacked',
    title: 'Unstacked.io',
    description: 'Unstacking the complexity of cloud.',
    url: 'https://unstacked.io',
    githubUrl: 'https://github.com/Unstacked',
    listDrafts: true
    // description ?
}

export const PAGE_SIZE = 8
