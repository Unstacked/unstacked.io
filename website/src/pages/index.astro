---
import Layout from '@layouts/Layout.astro';
import Card from '@components/Card.astro';
---

<Layout title="Welcome to Unstacked.">
	<main>
		<h1>Welcome to <span class="text-gradient">Unstacked</span></h1>
		<p class="instructions">
			Check out the links below for more about Unstacked
		</p>
		<ul role="list" class="link-card-grid">
			<Card
				href="https://www.youtube.com/channel/UCW7-QiS_ihP3JYTUSBQ8lRQ"
				title="Youtube"
				body="Watch Unstacked videos to learn about all things cloud"
			/>
			<Card
				href="https://github.com/Unstacked"
				title="Github"
				body="Explore projects that Unstacked are working on"
			/>
			<Card
				href="https://twitter.com/UnstackedCloud"
				title="Twitter"
				body="The short ramblings of whoever controls the twitter account"
			/>
			<!-- <Card
				href="/about.html"
				title="Chat"
				body="Come say hi to our amazing Discord community. ❤️"
			/> -->
		</ul>
	</main>
</Layout>

<style>
	:root {
		--astro-gradient: linear-gradient(0deg, #4f39fa, #da62c4);
	}

	h1 {
		margin: 2rem 0;
	}

	main {
		margin: auto;
		padding: 1em;
		max-width: 60ch;
	}

	.text-gradient {
		font-weight: 800;
		background-image: var(--astro-gradient);
		-webkit-background-clip: text;
		-webkit-text-fill-color: transparent;
		background-size: 100% 200%;
		background-position-y: 100%;
		border-radius: 0.4rem;
		animation: pulse 4s ease-in-out infinite;
	}

	@keyframes pulse {
		0%,
		100% {
			background-position-y: 0%;
		}
		50% {
			background-position-y: 80%;
		}
	}

	.instructions {
		line-height: 1.6;
		margin: 1rem 0;
		background: #4f39fa;
		padding: 1rem;
		border-radius: 0.4rem;
		color: var(--color-bg);
	}

	.instructions code {
		font-size: 0.875em;
		border: 0.1em solid var(--color-border);
		border-radius: 4px;
		padding: 0.15em 0.25em;
	}

	.link-card-grid {
		display: grid;
		grid-template-columns: repeat(auto-fit, minmax(24ch, 1fr));
		gap: 1rem;
		padding: 0;
	}
</style>
