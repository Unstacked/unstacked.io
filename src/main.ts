import { join } from 'path';
import { App, Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { static_site } from 'pwed-cdk';
import { Account } from './constant';

export class UnstackedIoSite extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    new static_site.StaticSite(this, 'UnstackedIoSite', {
      domain: 'unstacked.io',
      alternativeDomains: [
        'unstacked.cloud',
        'unstacked.xyz',
        'unstacked.me',
        'unstacked.media',
        'unstacked.online',
      ],
      path: join(__dirname, '..', 'website', 'dist'),
      enablePrettyPaths: true,
    });
  }
}

const env = {
  account: Account.WORKLOAD,
  region: 'us-east-1',
};

const app = new App();

new UnstackedIoSite(app, 'UnstackedIoWebsite', { env: env });

app.synth();
