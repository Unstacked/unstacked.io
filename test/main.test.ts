import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { Account } from '../src/constant';
import { UnstackedIoSite } from '../src/main';

test('Snapshot', () => {
  const app = new App();
  const stack = new UnstackedIoSite(app, 'test', {
    env: {
      account: Account.WORKLOAD,
      region: 'us-east-1',
    },
  });

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
