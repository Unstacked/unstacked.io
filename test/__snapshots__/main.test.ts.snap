// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Mappings": {
    "AWSCloudFrontPartitionHostedZoneIdMap": {
      "aws": {
        "zoneId": "Z2FDTNDATAQYW2",
      },
      "aws-cn": {
        "zoneId": "Z3RFFRIM2A3IF5",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": {
      "DependsOn": [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-806124249357-us-east-1",
          "S3Key": "f98b78092dcdd31f5e6d47489beb5f804d4835ef86a8085d0a2053cb9ae711da.zip",
        },
        "Handler": "index.handler",
        "Layers": [
          {
            "Ref": "UnstackedIoSiteStaticDeploymentAwsCliLayerE1C96417",
          },
        ],
        "Role": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-806124249357-us-east-1",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-806124249357-us-east-1/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "UnstackedIoSiteBucket9D979F83",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "UnstackedIoSiteBucket9D979F83",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": [
          {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "UnstackedIoSiteAAliasRecord51CF164B": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.io.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAAliasRecordunstackedcloudDE5679DD": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.cloud.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAAliasRecordunstackedmeEFAC4AEF": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.me.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAAliasRecordunstackedmedia30491CDF": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.media.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAAliasRecordunstackedonline93E8A041": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.online.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAAliasRecordunstackedxyz552E6AA2": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.xyz.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAaaaAliasRecord60CB0D8F": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.io.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAaaaAliasRecordunstackedcloudF35B8502": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.cloud.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAaaaAliasRecordunstackedme9B044338": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.me.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAaaaAliasRecordunstackedmedia91E32FE3": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.media.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAaaaAliasRecordunstackedonlineB4744925": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.online.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteAaaaAliasRecordunstackedxyz6776B9AE": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "UnstackedIoSiteDistributionD915FDDC",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "unstacked.xyz.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "UnstackedIoSiteBucket9D979F83": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "LifecycleConfiguration": {
          "Rules": [
            {
              "Status": "Enabled",
              "Transitions": [
                {
                  "StorageClass": "INTELLIGENT_TIERING",
                  "TransitionInDays": 7,
                },
              ],
            },
          ],
        },
        "Tags": [
          {
            "Key": "aws-cdk:cr-owned:573a57a8",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "UnstackedIoSiteBucketPolicy238796E0": {
      "Properties": {
        "Bucket": {
          "Ref": "UnstackedIoSiteBucket9D979F83",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": {
                "CanonicalUser": {
                  "Fn::GetAtt": [
                    "UnstackedIoSiteOriginAccessIdentityC4ED8DBA",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "UnstackedIoSiteBucket9D979F83",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "UnstackedIoSiteBucket9D979F83",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": {
                "CanonicalUser": {
                  "Fn::GetAtt": [
                    "UnstackedIoSiteOriginAccessIdentityC4ED8DBA",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "UnstackedIoSiteBucket9D979F83",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "UnstackedIoSiteCertificate977BE0E7": {
      "Properties": {
        "DomainName": "unstacked.io",
        "DomainValidationOptions": [
          {
            "DomainName": "unstacked.io",
            "HostedZoneId": "DUMMY",
          },
          {
            "DomainName": "unstacked.cloud",
            "HostedZoneId": "DUMMY",
          },
          {
            "DomainName": "unstacked.xyz",
            "HostedZoneId": "DUMMY",
          },
          {
            "DomainName": "unstacked.me",
            "HostedZoneId": "DUMMY",
          },
          {
            "DomainName": "unstacked.media",
            "HostedZoneId": "DUMMY",
          },
          {
            "DomainName": "unstacked.online",
            "HostedZoneId": "DUMMY",
          },
        ],
        "SubjectAlternativeNames": [
          "unstacked.cloud",
          "unstacked.xyz",
          "unstacked.me",
          "unstacked.media",
          "unstacked.online",
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
    },
    "UnstackedIoSiteDistributionD915FDDC": {
      "Properties": {
        "DistributionConfig": {
          "Aliases": [
            "unstacked.io",
            "unstacked.cloud",
            "unstacked.xyz",
            "unstacked.me",
            "unstacked.media",
            "unstacked.online",
          ],
          "CustomErrorResponses": [
            {
              "ErrorCode": 404,
              "ResponseCode": 404,
              "ResponsePagePath": "/404.html",
            },
          ],
          "DefaultCacheBehavior": {
            "AllowedMethods": [
              "GET",
              "HEAD",
              "OPTIONS",
            ],
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "FunctionAssociations": [
              {
                "EventType": "viewer-request",
                "FunctionARN": {
                  "Fn::GetAtt": [
                    "UnstackedIoSitePrettyPathHandler617786B0",
                    "FunctionARN",
                  ],
                },
              },
            ],
            "TargetOriginId": "testUnstackedIoSiteDistributionOrigin146207A76",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2and3",
          "IPV6Enabled": true,
          "Origins": [
            {
              "DomainName": {
                "Fn::GetAtt": [
                  "UnstackedIoSiteBucket9D979F83",
                  "RegionalDomainName",
                ],
              },
              "Id": "testUnstackedIoSiteDistributionOrigin146207A76",
              "S3OriginConfig": {
                "OriginAccessIdentity": {
                  "Fn::Join": [
                    "",
                    [
                      "origin-access-identity/cloudfront/",
                      {
                        "Ref": "UnstackedIoSiteOriginAccessIdentityC4ED8DBA",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "ViewerCertificate": {
            "AcmCertificateArn": {
              "Ref": "UnstackedIoSiteCertificate977BE0E7",
            },
            "MinimumProtocolVersion": "TLSv1.2_2021",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "UnstackedIoSiteOriginAccessIdentityC4ED8DBA": {
      "Properties": {
        "CloudFrontOriginAccessIdentityConfig": {
          "Comment": "Allows CloudFront to reach the bucket",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "UnstackedIoSitePrettyPathHandler617786B0": {
      "Properties": {
        "AutoPublish": true,
        "FunctionCode": ""use strict";
function handler(event) {
    var request = event.request;
    var uri = request.uri;
    // Check whether the URI is missing a file name.
    if (uri.endsWith('/')) {
        request.uri += 'index.html';
    }
    // Check whether the URI is missing a file extension.
    else if (!uri.includes('.')) {
        request.uri += '/index.html';
    }
    return request;
}
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoicHJldHR5UGF0aEhhbmRsZXIuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvbGliL3N0YXRpYy1zaXRlL3ByZXR0eVBhdGhIYW5kbGVyLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7QUFJQSxTQUFTLE9BQU8sQ0FBQyxLQUFtQjtJQUNsQyxJQUFJLE9BQU8sR0FBRyxLQUFLLENBQUMsT0FBTyxDQUFDO0lBQzVCLElBQUksR0FBRyxHQUFHLE9BQU8sQ0FBQyxHQUFHLENBQUM7SUFFdEIsZ0RBQWdEO0lBQ2hELElBQUksR0FBRyxDQUFDLFFBQVEsQ0FBQyxHQUFHLENBQUMsRUFBRTtRQUNyQixPQUFPLENBQUMsR0FBRyxJQUFJLFlBQVksQ0FBQztLQUM3QjtJQUNELHFEQUFxRDtTQUNoRCxJQUFJLENBQUMsR0FBRyxDQUFDLFFBQVEsQ0FBQyxHQUFHLENBQUMsRUFBRTtRQUMzQixPQUFPLENBQUMsR0FBRyxJQUFJLGFBQWEsQ0FBQztLQUM5QjtJQUVELE9BQU8sT0FBTyxDQUFDO0FBQ2pCLENBQUMiLCJzb3VyY2VzQ29udGVudCI6WyJpbnRlcmZhY2UgcmVxdWVzdEV2ZW50IHtcbiAgcmVxdWVzdDogeyB1cmk6IHN0cmluZyB9O1xufVxuXG5mdW5jdGlvbiBoYW5kbGVyKGV2ZW50OiByZXF1ZXN0RXZlbnQpIHtcbiAgdmFyIHJlcXVlc3QgPSBldmVudC5yZXF1ZXN0O1xuICB2YXIgdXJpID0gcmVxdWVzdC51cmk7XG5cbiAgLy8gQ2hlY2sgd2hldGhlciB0aGUgVVJJIGlzIG1pc3NpbmcgYSBmaWxlIG5hbWUuXG4gIGlmICh1cmkuZW5kc1dpdGgoJy8nKSkge1xuICAgIHJlcXVlc3QudXJpICs9ICdpbmRleC5odG1sJztcbiAgfVxuICAvLyBDaGVjayB3aGV0aGVyIHRoZSBVUkkgaXMgbWlzc2luZyBhIGZpbGUgZXh0ZW5zaW9uLlxuICBlbHNlIGlmICghdXJpLmluY2x1ZGVzKCcuJykpIHtcbiAgICByZXF1ZXN0LnVyaSArPSAnL2luZGV4Lmh0bWwnO1xuICB9XG5cbiAgcmV0dXJuIHJlcXVlc3Q7XG59XG4iXX0=",
        "FunctionConfig": {
          "Comment": "us-east-1testUnstackedIoSitePrettyPathHandlerA5C72912",
          "Runtime": "cloudfront-js-1.0",
        },
        "Name": "us-east-1testUnstackedIoSitePrettyPathHandlerA5C72912",
      },
      "Type": "AWS::CloudFront::Function",
    },
    "UnstackedIoSiteStaticDeploymentAwsCliLayerE1C96417": {
      "Properties": {
        "Content": {
          "S3Bucket": "cdk-hnb659fds-assets-806124249357-us-east-1",
          "S3Key": "39ee629321f531fffd853b944b2d6f3fa7b5276431c9a4fd4dc681303ab15080.zip",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "UnstackedIoSiteStaticDeploymentCustomResource0B456FBB": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "DestinationBucketName": {
          "Ref": "UnstackedIoSiteBucket9D979F83",
        },
        "DistributionId": {
          "Ref": "UnstackedIoSiteDistributionD915FDDC",
        },
        "DistributionPaths": [
          "/.hashfile",
          "/*",
        ],
        "Prune": true,
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": [
          "cdk-hnb659fds-assets-806124249357-us-east-1",
        ],
        "SourceObjectKeys": [
          "04a1a96a10e892598074c16d4e6674ba5c5debd62aac21553f4d3fffb5eef973.zip",
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
